.github/workflows/build.yml

```diff
@@ -0,0 +1,111 @@
+        default: ''
+        type: string
+      config:
+        required: false
+        default: 'Release'
+        type: string
+        # options:
+        #   - Debug
+        #   - Release
+      os:
+        required: false
+        type: string
+        default: '2022'
+        # options:
+        #   - 2019
+        #   - 2022
+      arch:
+        required: false
+        default: 'x64'
+        type: string
+        # options:
+        #   - x64
+      codeql:
+        required: false
+        default: false
+        type: bool
+        # options:
+        #   - true
+        #   - false
+
+
+permissions:
+  actions: read
+  contents: read
+  if: inputs.codeql
+    security-events: write  # For CodeQL
+
+jobs:
+  build:
+    name: Build
+    strategy:
+      fail-fast: false
+    runs-on: windows-${{ inputs.os }}
+    permissions:
+      actions: read
+      contents: read
+      security-events: write  # For CodeQL
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11
+      with:
+        submodules: recursive
+    - name: Initialize CodeQL
+      if: ${{ github.event_name == 'schedule' }}
+      uses: github/codeql-action/init@v3
+      with:
+        languages: c-cpp
+    - name: Setup MSBuild.exe
+      uses: microsoft/setup-msbuild@ab534842b4bdf384b8aaf93765dc6f721d9f5fab
+    - name: Prepare Machine
+      shell: PowerShell
+      run: tools/prepare-machine.ps1 -ForBuild -Verbose
+    - name: Install LLVM 11.0
+      run: |
+        choco install -y llvm --version 11.0.1 --allow-downgrade
+        echo "C:\Program Files\LLVM\bin" | Out-File -FilePath $env:GITHUB_PATH -Encoding utf8 -Append
+    - name: Nuget Restore
+      run: nuget.exe restore xdp.sln -ConfigFile src/nuget.config
+    - name: Nuget Restore (Installer)
+      run: nuget.exe restore src\xdpinstaller\xdpinstaller.sln -ConfigFile src/nuget.config
+    - name: Prepare for compiling eBPF programs
+      run: tools/prepare-machine.ps1 -ForEbpfBuild -Verbose
+    - name: Build
+      run: msbuild xdp.sln /m /p:configuration=${{ inputs.config }} /p:platform=${{ inputs.arch }}
+    - name: Sign Binaries
+      shell: PowerShell
+      run: tools/sign.ps1 -Config ${{ inputs.config }} -Arch ${{ inputs.arch }}
+    - name: Build Installer
+      shell: PowerShell
+      run: tools/create-installer.ps1 -Config ${{ inputs.config }} -Platform ${{ inputs.arch }}
+    - name: Upload Artifacts
+      if: inputs.os == 2022
+      uses: actions/upload-artifact@5d5d22a31266ced268874388b861e4b58bb5c2f3
+      with:
+        name: bin_${{ inputs.config }}_${{ inputs.arch }}
+        path: |
+          artifacts/bin
+          !artifacts/bin/**/*.ilk
+          !artifacts/bin/**/*.exp
+          !artifacts/bin/**/*.lastcodeanalysissucceeded
+    - name: Publish Symbols
+      uses: microsoft/action-publish-symbols@719c40b80e38bca806f3e01e1e3dd3a67554cd68
+      if: github.event_name != 'pull_request' && inputs.os == 2022
```

Comment:
Let's push this out as a workflow input. We won't want to publish when netperf uses this.